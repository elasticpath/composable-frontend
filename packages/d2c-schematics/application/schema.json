{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "SchematicsAngularApp",
  "title": "Angular Application Options Schema",
  "type": "object",
  "description": "Generates a new basic application definition in the \"projects\" subfolder of the workspace.",
  "additionalProperties": false,
  "properties": {
    "projectRoot": {
      "description": "The root directory of the new app.",
      "type": "string",
      "visible": false
    },
    "name": {
      "description": "The name of the new app.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the application?"
    },
    "skipTests": {
      "description": "Do not create \"spec.ts\" test files for the application.",
      "type": "boolean",
      "default": false,
      "alias": "S"
    }
  },
  "required": ["name"]
}
