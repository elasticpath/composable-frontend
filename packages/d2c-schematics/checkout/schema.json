{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "SchematicsCheckout",
  "title": "Checkout Options Schema",
  "type": "object",
  "description": "Generates checkout page and supporting functionality in the workspace.",
  "additionalProperties": true,
  "properties": {
    "skipTests": {
      "description": "Do not create \"spec.ts\" test files for the application.",
      "type": "boolean",
      "default": false,
      "alias": "S"
    },
    "skipConfig": {
      "description": "Do not configure Checkout in Elastic Path Commerce Cloud.",
      "type": "boolean",
      "default": false
    },
    "directory": {
      "type": "string",
      "description": "The directory name to create the workspace in."
    },
    "path": {
      "type": "string",
      "format": "path",
      "$default": {
        "$source": "workingDirectory"
      },
      "description": "The path at which to create the component file, relative to the current workspace. Default is a folder with the same name as the component in the project root.",
      "visible": false
    },
    "paymentGatewayType": {
      "description": "The type of payment gateway to use for your checkout experience.",
      "type": "string",
      "enum": ["EP Payments", "None"],
      "x-prompt": "Which payment gateway do you want to use?"
    },
    "epccClientId": {
      "description": "The client id value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "visible": "false"
    },
    "epccClientSecret": {
      "description": "The client secret value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "visible": "false"
    },
    "epccEndpointUrl": {
      "description": "The geographical endpoint url for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "visible": "false"
    }
  },
  "required": ["epccClientId", "epccClientSecret", "epccEndpointUrl", "directory"]
}
