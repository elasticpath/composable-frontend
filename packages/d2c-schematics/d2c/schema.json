{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "SchematicsD2C",
  "title": "Elastic Path Commerce Cloud Direct to Consumer (D2C) Options Schema",
  "type": "object",
  "description": "Creates a new Elastic Path Commerce Cloud storefront by combining the workspace and application schematics.",
  "additionalProperties": true,
  "properties": {
    "directory": {
      "type": "string",
      "description": "The directory name to create the workspace in."
    },
    "name": {
      "description": "The name of the new workspace and project.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the new workspace and initial project?"
    },
    "packageManager": {
      "description": "Node package manager to use.",
      "type": "string",
      "enum": ["npm", "yarn", "pnpm"],
      "default": "npm"
    },
    "epccClientId": {
      "description": "The client id value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "x-prompt": "What is the client id for your store?"
    },
    "epccClientSecret": {
      "description": "The client secret value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "x-prompt": "What is the client secret for your store?"
    },
    "epccEndpointUrl": {
      "description": "The geographical endpoint url for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "default": "api.moltin.com",
      "x-prompt": "What is the endpoint for your store?"
    },
    "skipInstall": {
      "description": "Do not install dependency packages.",
      "type": "boolean",
      "default": false
    },
    "skipGit": {
      "description": "Do not initialize a git repository.",
      "type": "boolean",
      "default": false,
      "alias": "g"
    },
    "commit": {
      "description": "Initial git repository commit information.",
      "oneOf": [
        { "type": "boolean" },
        {
          "type": "object",
          "properties": {
            "name": {
              "type": "string"
            },
            "email": {
              "type": "string",
              "format": "email"
            },
            "message": {
              "type": "string"
            }
          },
          "required": ["name", "email"]
        }
      ],
      "default": true
    },
    "skipTests": {
      "description": "Do not generate \"spec.ts\" test files for the new project.",
      "type": "boolean",
      "default": false,
      "alias": "S"
    },
    "plpType": {
      "description": "The type of product list page to use.",
      "type": "string",
      "enum": ["Algolia", "None"]
    }
  },
  "required": ["name", "epccClientId", "epccClientSecret", "epccEndpointUrl"]
}
