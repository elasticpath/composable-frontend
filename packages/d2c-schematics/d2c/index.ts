import {
  Rule,
  SchematicContext,
  Tree,
  apply,
  chain,
  empty,
  mergeWith,
  move,
  schematic,
} from "@angular-devkit/schematics"
import {
  NodePackageInstallTask,
  RepositoryInitializerTask,
} from "@angular-devkit/schematics/tasks"
import { Schema as ApplicationOptions } from "../application/schema"
import { Schema as WorkspaceOptions } from "../workspace/schema"
import { Schema as ProductListOptions } from "../product-list-page/schema"
import { Schema as D2COptions } from "./schema"

export default function (options: D2COptions): Rule {
  if (!options.directory) {
    // If scoped project (i.e. "@foo/bar"), convert directory to "foo/bar".
    options.directory = options.name.startsWith("@")
      ? options.name.slice(1)
      : options.name
  }

  const projectRoot = ""

  const {
    epccEndpointUrl,
    epccClientSecret,
    epccClientId,
    plpType,
    skipTests,
    name,
  } = options

  const workspaceOptions: WorkspaceOptions = {
    name,
    epccClientId,
    epccClientSecret,
    epccEndpointUrl,
  }

  const applicationOptions: ApplicationOptions = {
    projectRoot,
    name,
    skipTests,
  }

  const plpOptions: ProductListOptions = {
    ...options,
    path: projectRoot,
    skipTests,
    epccClientId,
    epccClientSecret,
    epccEndpointUrl,
    plpType,
    directory: options.directory,
  }

  return chain([
    mergeWith(
      apply(empty(), [
        schematic("workspace", workspaceOptions),
        schematic("application", applicationOptions),
        schematic("cart", {
          path: projectRoot,
        }),
        schematic("header", {
          path: projectRoot,
          search: false,
        }),
        schematic("footer", {
          path: projectRoot,
        }),
        schematic("pdp", {
          path: projectRoot,
        }),
        schematic("plp", plpOptions),
        schematic("home", {
          path: projectRoot,
        }),
        move(options.directory),
      ])
    ),
    (_host: Tree, context: SchematicContext) => {
      let packageTask
      if (!options.skipGit) {
        const commit =
          typeof options.commit == "object"
            ? options.commit
            : options.commit
            ? {}
            : false

        packageTask = context.addTask(
          new RepositoryInitializerTask(options.directory, commit)
        )
      }
      if (!options.skipInstall) {
        context.addTask(
          new NodePackageInstallTask({
            workingDirectory: options.directory,
            packageManager: "yarn",
          }),
          packageTask ? [packageTask] : []
        )
      }
    },
  ])
}
