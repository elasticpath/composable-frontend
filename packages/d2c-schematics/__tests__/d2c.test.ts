/**
 * @license
 * Copyright Google LLC All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */

import { SchematicTestRunner } from "@angular-devkit/schematics/testing"
import path from "path"
import { Schema as D2COptions } from "../d2c/schema"

const collectionPath = path.join(__dirname, "../collection.json")

describe("EP New Schematic", () => {
  const schematicRunner = new SchematicTestRunner("schematics", collectionPath)
  const defaultOptions: D2COptions = {
    name: "foo",
    directory: "bar",
    epccClientId: "123",
    epccClientSecret: "456",
    epccEndpointUrl: "euwest.api.elasticpath.com",
  }

  it("should create files of a workspace", async () => {
    const options = { ...defaultOptions }

    const tree: any = await schematicRunner
      .runSchematicAsync("d2c", options)
      .toPromise()
      .catch((err) => console.log("error caught:", err))

    const files = tree.files
    expect(files).toContain("/bar/package.json")
  })

  it("should create cart schematic files", async () => {
    const options = { ...defaultOptions }
    const tree = await schematicRunner
      .runSchematicAsync("d2c", options)
      .toPromise()
    const files = tree.files

    expect(files).toIncludeAllPartialMembers([
      "/bar/src/lib/custom-rule-headers.ts",
      "/bar/src/lib/ep-client-store.ts",
      "/bar/src/lib/resolve-cart-env.ts",
      "/bar/src/app/cart/page.tsx",
      "/bar/src/lib/build-site-navigation.ts",
      "/bar/src/services/hierarchy.ts",
      "/bar/src/services/cart.ts",
      "/bar/src/lib/epcc-implicit-client.ts",
      "/bar/src/lib/types/store-context.ts",
    ])
  })
})
