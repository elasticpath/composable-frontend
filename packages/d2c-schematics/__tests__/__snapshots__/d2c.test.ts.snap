// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EP New Schematic should create cart schematic files 1`] = `
[
  "/bar/README.md",
  "/bar/.composablerc",
  "/bar/.env.example",
  "/bar/.env.local",
  "/bar/.env.test",
  "/bar/.eslintrc.json",
  "/bar/.gitignore",
  "/bar/.lintstagedrc.js",
  "/bar/.prettierignore",
  "/bar/.prettierrc",
  "/bar/license.md",
  "/bar/next-env.d.ts",
  "/bar/next.config.js",
  "/bar/package.json",
  "/bar/playwright.config.ts",
  "/bar/postcss.config.js",
  "/bar/tailwind.config.ts",
  "/bar/tsconfig.json",
  "/bar/vite.config.ts",
  "/bar/public/favicon.ico",
  "/bar/e2e/product-details-page.spec.ts",
  "/bar/e2e/product-list-page.spec.ts",
  "/bar/e2e/home-page.spec.ts",
  "/bar/e2e/util/epcc-admin-client.ts",
  "/bar/e2e/util/epcc-client.ts",
  "/bar/e2e/util/fill-form-field.ts",
  "/bar/e2e/util/has-published-catalog.ts",
  "/bar/e2e/util/missing-published-catalog.ts",
  "/bar/e2e/util/skip-ci-env.ts",
  "/bar/e2e/util/get-cart-id.ts",
  "/bar/e2e/util/resolver-product-from-store.ts",
  "/bar/e2e/models/d2c-cart-page.ts",
  "/bar/e2e/models/d2c-product-detail-page.ts",
  "/bar/e2e/models/d2c-home-page.ts",
  "/bar/src/middleware.ts",
  "/bar/src/app/error.tsx",
  "/bar/src/app/layout.tsx",
  "/bar/src/app/not-found.tsx",
  "/bar/src/app/providers.tsx",
  "/bar/src/app/(store)/layout.tsx",
  "/bar/src/app/(store)/not-found.tsx",
  "/bar/src/app/(store)/page.tsx",
  "/bar/src/app/(store)/about/page.tsx",
  "/bar/src/app/(store)/faq/page.tsx",
  "/bar/src/app/(store)/shipping/page.tsx",
  "/bar/src/app/(store)/terms/page.tsx",
  "/bar/src/app/(store)/cart/CartItem.tsx",
  "/bar/src/app/(store)/cart/CartItemWide.tsx",
  "/bar/src/app/(store)/cart/CartSidebar.tsx",
  "/bar/src/app/(store)/cart/CartView.tsx",
  "/bar/src/app/(store)/cart/YourBag.tsx",
  "/bar/src/app/(store)/cart/page.tsx",
  "/bar/src/app/(store)/products/[productId]/page.tsx",
  "/bar/src/app/(store)/products/[productId]/product-display.tsx",
  "/bar/src/app/(store)/search/search.tsx",
  "/bar/src/app/(store)/search/[[...node]]/layout.tsx",
  "/bar/src/app/(store)/search/[[...node]]/page.tsx",
  "/bar/src/app/(store)/account/AccountNavigation.tsx",
  "/bar/src/app/(store)/account/breadcrumb.tsx",
  "/bar/src/app/(store)/account/layout.tsx",
  "/bar/src/app/(store)/account/addresses/actions.ts",
  "/bar/src/app/(store)/account/addresses/page.tsx",
  "/bar/src/app/(store)/account/addresses/[addressId]/page.tsx",
  "/bar/src/app/(store)/account/addresses/add/page.tsx",
  "/bar/src/app/(store)/account/orders/OrderItem.tsx",
  "/bar/src/app/(store)/account/orders/OrderItemWithDetails.tsx",
  "/bar/src/app/(store)/account/orders/page.tsx",
  "/bar/src/app/(store)/account/orders/[orderId]/OrderLineItem.tsx",
  "/bar/src/app/(store)/account/orders/[orderId]/ProductThumbnail.tsx",
  "/bar/src/app/(store)/account/orders/[orderId]/page.tsx",
  "/bar/src/app/(store)/account/summary/YourInfoForm.tsx",
  "/bar/src/app/(store)/account/summary/actions.ts",
  "/bar/src/app/(store)/account/summary/page.tsx",
  "/bar/src/app/configuration-error/page.tsx",
  "/bar/src/app/(auth)/account-memeber-credentials-schema.ts",
  "/bar/src/app/(auth)/actions.ts",
  "/bar/src/app/(auth)/layout.tsx",
  "/bar/src/app/(auth)/not-found.tsx",
  "/bar/src/app/(auth)/login/LoginForm.tsx",
  "/bar/src/app/(auth)/login/page.tsx",
  "/bar/src/app/(auth)/register/page.tsx",
  "/bar/src/components/Checkbox.tsx",
  "/bar/src/components/LoadingDots.tsx",
  "/bar/src/components/Spinner.tsx",
  "/bar/src/components/shimmer.tsx",
  "/bar/src/components/NoImage.tsx",
  "/bar/src/components/breadcrumb.tsx",
  "/bar/src/components/accordion/Accordion.tsx",
  "/bar/src/components/alert/Alert.tsx",
  "/bar/src/components/button/Button.tsx",
  "/bar/src/components/button/FormStatusButton.tsx",
  "/bar/src/components/button/LoaderIcon.tsx",
  "/bar/src/components/button/StatusButton.tsx",
  "/bar/src/components/button/TextButton.tsx",
  "/bar/src/components/form/Form.tsx",
  "/bar/src/components/icons/ep-icon.tsx",
  "/bar/src/components/icons/ep-logo.tsx",
  "/bar/src/components/icons/github-icon.tsx",
  "/bar/src/components/icons/cart.tsx",
  "/bar/src/components/input/Input.tsx",
  "/bar/src/components/label/Label.tsx",
  "/bar/src/components/navigation-menu/NavigationMenu.tsx",
  "/bar/src/components/number-input/EditQuantityButton.tsx",
  "/bar/src/components/number-input/NumberInput.tsx",
  "/bar/src/components/number-input/use-counter.ts",
  "/bar/src/components/pagination/Pagination.tsx",
  "/bar/src/components/pagination/ResourcePagination.tsx",
  "/bar/src/components/radio-group/RadioGroup.tsx",
  "/bar/src/components/select/Select.tsx",
  "/bar/src/components/separator/Separator.tsx",
  "/bar/src/components/shared/blurb.tsx",
  "/bar/src/components/sheet/Sheet.tsx",
  "/bar/src/components/skeleton/Skeleton.tsx",
  "/bar/src/components/toast/toaster.tsx",
  "/bar/src/components/cart/CartDiscounts.tsx",
  "/bar/src/components/cart/CartSheet.tsx",
  "/bar/src/components/header/AccountMobileMenu.tsx",
  "/bar/src/components/header/Header.tsx",
  "/bar/src/components/header/account/AccountMenu.tsx",
  "/bar/src/components/header/account/AccountSwitcher.tsx",
  "/bar/src/components/header/account/switch-button.tsx",
  "/bar/src/components/header/navigation/MobileAccountSwitcher.tsx",
  "/bar/src/components/header/navigation/MobileNavBar.tsx",
  "/bar/src/components/header/navigation/MobileNavBarButton.tsx",
  "/bar/src/components/header/navigation/NavBar.tsx",
  "/bar/src/components/header/navigation/NavBarPopover.tsx",
  "/bar/src/components/header/navigation/NavItemContent.tsx",
  "/bar/src/components/header/navigation/NavMenu.tsx",
  "/bar/src/components/footer/Footer.tsx",
  "/bar/src/components/product/Price.tsx",
  "/bar/src/components/product/ProductDetails.tsx",
  "/bar/src/components/product/ProductExtensions.tsx",
  "/bar/src/components/product/ProductSummary.tsx",
  "/bar/src/components/product/SimpleProduct.tsx",
  "/bar/src/components/product/StrikePrice.tsx",
  "/bar/src/components/product/bundles/BundleProduct.tsx",
  "/bar/src/components/product/bundles/ProductComponent.tsx",
  "/bar/src/components/product/bundles/ProductComponents.tsx",
  "/bar/src/components/product/bundles/form-parsers.test.ts",
  "/bar/src/components/product/bundles/form-parsers.ts",
  "/bar/src/components/product/bundles/sort-by-order.ts",
  "/bar/src/components/product/bundles/validation-schema.test.ts",
  "/bar/src/components/product/bundles/validation-schema.ts",
  "/bar/src/components/product/carousel/CarouselListener.tsx",
  "/bar/src/components/product/carousel/HorizontalCarousel.tsx",
  "/bar/src/components/product/carousel/ProductCarousel.module.css",
  "/bar/src/components/product/carousel/ProductCarousel.tsx",
  "/bar/src/components/product/carousel/ProductHighlightCarousel.tsx",
  "/bar/src/components/product/variations/ProductVariationColor.tsx",
  "/bar/src/components/product/variations/ProductVariationStandard.tsx",
  "/bar/src/components/product/variations/ProductVariations.tsx",
  "/bar/src/components/product/variations/VariationProduct.tsx",
  "/bar/src/components/search/Hit.tsx",
  "/bar/src/components/search/Hits.tsx",
  "/bar/src/components/search/MobileFilters.tsx",
  "/bar/src/components/search/NoResults.tsx",
  "/bar/src/components/search/NodeMenu.tsx",
  "/bar/src/components/search/Pagination.tsx",
  "/bar/src/components/search/ProductsProvider.tsx",
  "/bar/src/components/search/SearchResults.tsx",
  "/bar/src/components/featured-products/FeaturedProducts.tsx",
  "/bar/src/components/featured-products/fetchFeaturedProducts.ts",
  "/bar/src/components/promotion-banner/PromotionBanner.tsx",
  "/bar/src/lib/build-site-navigation.ts",
  "/bar/src/lib/cn.tsx",
  "/bar/src/lib/constants.ts",
  "/bar/src/lib/cookie-constants.ts",
  "/bar/src/lib/custom-rule-headers.ts",
  "/bar/src/lib/epcc-errors.ts",
  "/bar/src/lib/epcc-implicit-client.ts",
  "/bar/src/lib/epcc-server-client.ts",
  "/bar/src/lib/epcc-server-side-credentials-client.ts",
  "/bar/src/lib/epcc-server-side-implicit-client.ts",
  "/bar/src/lib/form-url-encode-body.ts",
  "/bar/src/lib/format-currency.tsx",
  "/bar/src/lib/format-iso-date-string.ts",
  "/bar/src/lib/get-error-message.ts",
  "/bar/src/lib/get-store-initial-state.ts",
  "/bar/src/lib/is-account-member-authenticated.ts",
  "/bar/src/lib/is-empty-object.ts",
  "/bar/src/lib/resolve-cart-env.ts",
  "/bar/src/lib/resolve-ep-currency-code.ts",
  "/bar/src/lib/resolve-epcc-env.ts",
  "/bar/src/lib/retrieve-account-member-credentials.ts",
  "/bar/src/lib/sort-alphabetically.ts",
  "/bar/src/lib/to-base-64.ts",
  "/bar/src/lib/token-expired.ts",
  "/bar/src/lib/cart-cookie-server.ts",
  "/bar/src/lib/color-lookup.ts",
  "/bar/src/lib/connect-products-with-main-images.ts",
  "/bar/src/lib/is-supported-extension.ts",
  "/bar/src/lib/product-context.ts",
  "/bar/src/lib/product-helper.test.ts",
  "/bar/src/lib/product-helper.ts",
  "/bar/src/lib/product-util.test.ts",
  "/bar/src/lib/product-util.ts",
  "/bar/src/lib/build-breadcrumb-lookup.ts",
  "/bar/src/lib/create-breadcrumb.ts",
  "/bar/src/lib/use-debounced.ts",
  "/bar/src/lib/file-lookup.test.ts",
  "/bar/src/lib/file-lookup.ts",
  "/bar/src/lib/middleware/apply-set-cookie.ts",
  "/bar/src/lib/middleware/cart-cookie-middleware.ts",
  "/bar/src/lib/middleware/create-missing-environment-variable-url.ts",
  "/bar/src/lib/middleware/implicit-auth-middleware.ts",
  "/bar/src/lib/middleware/middleware-runner.ts",
  "/bar/src/lib/types/deep-partial.ts",
  "/bar/src/lib/types/non-empty-array.ts",
  "/bar/src/lib/types/read-only-non-empty-array.ts",
  "/bar/src/lib/types/matrix-object-entry.ts",
  "/bar/src/lib/types/product-types.ts",
  "/bar/src/lib/types/breadcrumb-lookup.ts",
  "/bar/src/lib/types/unpacked.ts",
  "/bar/src/services/hierarchy.ts",
  "/bar/src/services/cart.ts",
  "/bar/src/services/products.ts",
  "/bar/src/styles/globals.css",
]
`;

exports[`EP New Schematic should create files of a workspace 1`] = `
[
  "/bar/README.md",
  "/bar/.composablerc",
  "/bar/.env.example",
  "/bar/.env.local",
  "/bar/.env.test",
  "/bar/.eslintrc.json",
  "/bar/.gitignore",
  "/bar/.lintstagedrc.js",
  "/bar/.prettierignore",
  "/bar/.prettierrc",
  "/bar/license.md",
  "/bar/next-env.d.ts",
  "/bar/next.config.js",
  "/bar/package.json",
  "/bar/playwright.config.ts",
  "/bar/postcss.config.js",
  "/bar/tailwind.config.ts",
  "/bar/tsconfig.json",
  "/bar/vite.config.ts",
  "/bar/public/favicon.ico",
  "/bar/e2e/product-details-page.spec.ts",
  "/bar/e2e/product-list-page.spec.ts",
  "/bar/e2e/home-page.spec.ts",
  "/bar/e2e/util/epcc-admin-client.ts",
  "/bar/e2e/util/epcc-client.ts",
  "/bar/e2e/util/fill-form-field.ts",
  "/bar/e2e/util/has-published-catalog.ts",
  "/bar/e2e/util/missing-published-catalog.ts",
  "/bar/e2e/util/skip-ci-env.ts",
  "/bar/e2e/util/get-cart-id.ts",
  "/bar/e2e/util/resolver-product-from-store.ts",
  "/bar/e2e/models/d2c-cart-page.ts",
  "/bar/e2e/models/d2c-product-detail-page.ts",
  "/bar/e2e/models/d2c-home-page.ts",
  "/bar/src/middleware.ts",
  "/bar/src/app/error.tsx",
  "/bar/src/app/layout.tsx",
  "/bar/src/app/not-found.tsx",
  "/bar/src/app/providers.tsx",
  "/bar/src/app/(store)/layout.tsx",
  "/bar/src/app/(store)/not-found.tsx",
  "/bar/src/app/(store)/page.tsx",
  "/bar/src/app/(store)/about/page.tsx",
  "/bar/src/app/(store)/faq/page.tsx",
  "/bar/src/app/(store)/shipping/page.tsx",
  "/bar/src/app/(store)/terms/page.tsx",
  "/bar/src/app/(store)/cart/CartItem.tsx",
  "/bar/src/app/(store)/cart/CartItemWide.tsx",
  "/bar/src/app/(store)/cart/CartSidebar.tsx",
  "/bar/src/app/(store)/cart/CartView.tsx",
  "/bar/src/app/(store)/cart/YourBag.tsx",
  "/bar/src/app/(store)/cart/page.tsx",
  "/bar/src/app/(store)/products/[productId]/page.tsx",
  "/bar/src/app/(store)/products/[productId]/product-display.tsx",
  "/bar/src/app/(store)/search/search.tsx",
  "/bar/src/app/(store)/search/[[...node]]/layout.tsx",
  "/bar/src/app/(store)/search/[[...node]]/page.tsx",
  "/bar/src/app/(store)/account/AccountNavigation.tsx",
  "/bar/src/app/(store)/account/breadcrumb.tsx",
  "/bar/src/app/(store)/account/layout.tsx",
  "/bar/src/app/(store)/account/addresses/actions.ts",
  "/bar/src/app/(store)/account/addresses/page.tsx",
  "/bar/src/app/(store)/account/addresses/[addressId]/page.tsx",
  "/bar/src/app/(store)/account/addresses/add/page.tsx",
  "/bar/src/app/(store)/account/orders/OrderItem.tsx",
  "/bar/src/app/(store)/account/orders/OrderItemWithDetails.tsx",
  "/bar/src/app/(store)/account/orders/page.tsx",
  "/bar/src/app/(store)/account/orders/[orderId]/OrderLineItem.tsx",
  "/bar/src/app/(store)/account/orders/[orderId]/ProductThumbnail.tsx",
  "/bar/src/app/(store)/account/orders/[orderId]/page.tsx",
  "/bar/src/app/(store)/account/summary/YourInfoForm.tsx",
  "/bar/src/app/(store)/account/summary/actions.ts",
  "/bar/src/app/(store)/account/summary/page.tsx",
  "/bar/src/app/configuration-error/page.tsx",
  "/bar/src/app/(auth)/account-memeber-credentials-schema.ts",
  "/bar/src/app/(auth)/actions.ts",
  "/bar/src/app/(auth)/layout.tsx",
  "/bar/src/app/(auth)/not-found.tsx",
  "/bar/src/app/(auth)/login/LoginForm.tsx",
  "/bar/src/app/(auth)/login/page.tsx",
  "/bar/src/app/(auth)/register/page.tsx",
  "/bar/src/components/Checkbox.tsx",
  "/bar/src/components/LoadingDots.tsx",
  "/bar/src/components/Spinner.tsx",
  "/bar/src/components/shimmer.tsx",
  "/bar/src/components/NoImage.tsx",
  "/bar/src/components/breadcrumb.tsx",
  "/bar/src/components/accordion/Accordion.tsx",
  "/bar/src/components/alert/Alert.tsx",
  "/bar/src/components/button/Button.tsx",
  "/bar/src/components/button/FormStatusButton.tsx",
  "/bar/src/components/button/LoaderIcon.tsx",
  "/bar/src/components/button/StatusButton.tsx",
  "/bar/src/components/button/TextButton.tsx",
  "/bar/src/components/form/Form.tsx",
  "/bar/src/components/icons/ep-icon.tsx",
  "/bar/src/components/icons/ep-logo.tsx",
  "/bar/src/components/icons/github-icon.tsx",
  "/bar/src/components/icons/cart.tsx",
  "/bar/src/components/input/Input.tsx",
  "/bar/src/components/label/Label.tsx",
  "/bar/src/components/navigation-menu/NavigationMenu.tsx",
  "/bar/src/components/number-input/EditQuantityButton.tsx",
  "/bar/src/components/number-input/NumberInput.tsx",
  "/bar/src/components/number-input/use-counter.ts",
  "/bar/src/components/pagination/Pagination.tsx",
  "/bar/src/components/pagination/ResourcePagination.tsx",
  "/bar/src/components/radio-group/RadioGroup.tsx",
  "/bar/src/components/select/Select.tsx",
  "/bar/src/components/separator/Separator.tsx",
  "/bar/src/components/shared/blurb.tsx",
  "/bar/src/components/sheet/Sheet.tsx",
  "/bar/src/components/skeleton/Skeleton.tsx",
  "/bar/src/components/toast/toaster.tsx",
  "/bar/src/components/cart/CartDiscounts.tsx",
  "/bar/src/components/cart/CartSheet.tsx",
  "/bar/src/components/header/AccountMobileMenu.tsx",
  "/bar/src/components/header/Header.tsx",
  "/bar/src/components/header/account/AccountMenu.tsx",
  "/bar/src/components/header/account/AccountSwitcher.tsx",
  "/bar/src/components/header/account/switch-button.tsx",
  "/bar/src/components/header/navigation/MobileAccountSwitcher.tsx",
  "/bar/src/components/header/navigation/MobileNavBar.tsx",
  "/bar/src/components/header/navigation/MobileNavBarButton.tsx",
  "/bar/src/components/header/navigation/NavBar.tsx",
  "/bar/src/components/header/navigation/NavBarPopover.tsx",
  "/bar/src/components/header/navigation/NavItemContent.tsx",
  "/bar/src/components/header/navigation/NavMenu.tsx",
  "/bar/src/components/footer/Footer.tsx",
  "/bar/src/components/product/Price.tsx",
  "/bar/src/components/product/ProductDetails.tsx",
  "/bar/src/components/product/ProductExtensions.tsx",
  "/bar/src/components/product/ProductSummary.tsx",
  "/bar/src/components/product/SimpleProduct.tsx",
  "/bar/src/components/product/StrikePrice.tsx",
  "/bar/src/components/product/bundles/BundleProduct.tsx",
  "/bar/src/components/product/bundles/ProductComponent.tsx",
  "/bar/src/components/product/bundles/ProductComponents.tsx",
  "/bar/src/components/product/bundles/form-parsers.test.ts",
  "/bar/src/components/product/bundles/form-parsers.ts",
  "/bar/src/components/product/bundles/sort-by-order.ts",
  "/bar/src/components/product/bundles/validation-schema.test.ts",
  "/bar/src/components/product/bundles/validation-schema.ts",
  "/bar/src/components/product/carousel/CarouselListener.tsx",
  "/bar/src/components/product/carousel/HorizontalCarousel.tsx",
  "/bar/src/components/product/carousel/ProductCarousel.module.css",
  "/bar/src/components/product/carousel/ProductCarousel.tsx",
  "/bar/src/components/product/carousel/ProductHighlightCarousel.tsx",
  "/bar/src/components/product/variations/ProductVariationColor.tsx",
  "/bar/src/components/product/variations/ProductVariationStandard.tsx",
  "/bar/src/components/product/variations/ProductVariations.tsx",
  "/bar/src/components/product/variations/VariationProduct.tsx",
  "/bar/src/components/search/Hit.tsx",
  "/bar/src/components/search/Hits.tsx",
  "/bar/src/components/search/MobileFilters.tsx",
  "/bar/src/components/search/NoResults.tsx",
  "/bar/src/components/search/NodeMenu.tsx",
  "/bar/src/components/search/Pagination.tsx",
  "/bar/src/components/search/ProductsProvider.tsx",
  "/bar/src/components/search/SearchResults.tsx",
  "/bar/src/components/featured-products/FeaturedProducts.tsx",
  "/bar/src/components/featured-products/fetchFeaturedProducts.ts",
  "/bar/src/components/promotion-banner/PromotionBanner.tsx",
  "/bar/src/lib/build-site-navigation.ts",
  "/bar/src/lib/cn.tsx",
  "/bar/src/lib/constants.ts",
  "/bar/src/lib/cookie-constants.ts",
  "/bar/src/lib/custom-rule-headers.ts",
  "/bar/src/lib/epcc-errors.ts",
  "/bar/src/lib/epcc-implicit-client.ts",
  "/bar/src/lib/epcc-server-client.ts",
  "/bar/src/lib/epcc-server-side-credentials-client.ts",
  "/bar/src/lib/epcc-server-side-implicit-client.ts",
  "/bar/src/lib/form-url-encode-body.ts",
  "/bar/src/lib/format-currency.tsx",
  "/bar/src/lib/format-iso-date-string.ts",
  "/bar/src/lib/get-error-message.ts",
  "/bar/src/lib/get-store-initial-state.ts",
  "/bar/src/lib/is-account-member-authenticated.ts",
  "/bar/src/lib/is-empty-object.ts",
  "/bar/src/lib/resolve-cart-env.ts",
  "/bar/src/lib/resolve-ep-currency-code.ts",
  "/bar/src/lib/resolve-epcc-env.ts",
  "/bar/src/lib/retrieve-account-member-credentials.ts",
  "/bar/src/lib/sort-alphabetically.ts",
  "/bar/src/lib/to-base-64.ts",
  "/bar/src/lib/token-expired.ts",
  "/bar/src/lib/cart-cookie-server.ts",
  "/bar/src/lib/color-lookup.ts",
  "/bar/src/lib/connect-products-with-main-images.ts",
  "/bar/src/lib/is-supported-extension.ts",
  "/bar/src/lib/product-context.ts",
  "/bar/src/lib/product-helper.test.ts",
  "/bar/src/lib/product-helper.ts",
  "/bar/src/lib/product-util.test.ts",
  "/bar/src/lib/product-util.ts",
  "/bar/src/lib/build-breadcrumb-lookup.ts",
  "/bar/src/lib/create-breadcrumb.ts",
  "/bar/src/lib/use-debounced.ts",
  "/bar/src/lib/file-lookup.test.ts",
  "/bar/src/lib/file-lookup.ts",
  "/bar/src/lib/middleware/apply-set-cookie.ts",
  "/bar/src/lib/middleware/cart-cookie-middleware.ts",
  "/bar/src/lib/middleware/create-missing-environment-variable-url.ts",
  "/bar/src/lib/middleware/implicit-auth-middleware.ts",
  "/bar/src/lib/middleware/middleware-runner.ts",
  "/bar/src/lib/types/deep-partial.ts",
  "/bar/src/lib/types/non-empty-array.ts",
  "/bar/src/lib/types/read-only-non-empty-array.ts",
  "/bar/src/lib/types/matrix-object-entry.ts",
  "/bar/src/lib/types/product-types.ts",
  "/bar/src/lib/types/breadcrumb-lookup.ts",
  "/bar/src/lib/types/unpacked.ts",
  "/bar/src/services/hierarchy.ts",
  "/bar/src/services/cart.ts",
  "/bar/src/services/products.ts",
  "/bar/src/styles/globals.css",
]
`;
